PR to address incorrect package versions in DataProtection Configuration document (Issue #19770)

markmcgookin opened the pull request:
Fixes #19770

Update to fix package references and minor configuration changes when moving to the new libraries.

Left the workaround in place for the issue with the old library incase people are googling the error and come across the document (included a comment suggesting them to upgrade)

guardrex commented:
@Rick-Anderson ... This isn't the only spot for the updates. I don't see any sample apps that need work, but there are perhaps four more topics that need work.

Rick-Anderson commented:
@pakrym please review or recommend a reviewer

pakrym commented on aspnetcore/security/data-protection/configuration/overview.md:
```diff
@@ -40,9 +40,12 @@ To store keys in [Azure Key Vault](https://azure.microsoft.com/services/key-vaul
     services.AddDataProtection()
-        .PersistKeysToAzureBlobStorage(new Uri("<blobUriWithSasToken>"))
-        .ProtectKeysWithAzureKeyVault("<keyIdentifier>", "<clientId>", "<clientSecret>");
+                    .PersistKeysToAzureBlobStorage("<storageAccountConnectionString>", "<storageContainerName>", "<keyStorageBlobName>")
```
The overload that takes blob URI still exists, I would recommend keeping using it along with the `DefaultAzureCredential` from `Azure.Identity`

markmcgookin replied:
Ok, i've rolled that back to the original overload. I must have missed it when setting up my project and moved over to the other one.

markmcgookin commented:
I've updated the PR at @pakrym 's request to have the original setup arguments, I changed mine when migrating to the new libraries, but apparently the original ones in the documentation still work.

guardrex reviewed (approved):
Thanks @markmcgookin! :rocket:
